### PR TITLE
Add Lightbend's SBR to Akka Cluster, #29085

### PR DESCRIPTION
**Pull request by patriknw**
* change package name to akka.cluster.sbr
* reference.conf has same config paths
* akka.cluster.sbr.SplitBrainResolverProvider instead of com.lightbend.akka.sbr.SplitBrainResolverProvider
* dependency from akka-cluster to akka-coordination, for lease strategy
* move TestLease to akka-coordination and use that in SBR tests
* remove keep-referee strategy
* use keep-majority by default
* review and adjust reference documentation

References #29085

See individual commits if you want to review the minor changes from Lightbend SBR, which corresponds to the first commit.

It is compatible with Lightbend SBR also for rolling updates.

### CODE REVIEW

File: akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sbr/SplitBrainResolverIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+import akka.testkit.ImplicitSender
+import akka.testkit.LongRunningTest
+import akka.testkit.TestKit
+import akka.testkit.TestProbe
+import akka.util.Timeout
+
+/*
+ * Depends on akka private classes so needs to be in this package
+ */
+object SplitBrainResolverIntegrationSpec extends MultiNodeConfig {
```
Comment (patriknw) [inline akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sbr/SplitBrainResolverIntegrationSpec.scala:39]: We should be aware of that the additional SBR tests add 12 minutes to the jenkins multi-node jobs.

File: akka-docs/src/main/paradox/common/other-modules.md [hunk 2]
```diff
@@ -26,15 +26,14 @@ An Akka Persistence journal and snapshot store backed by Couchbase.
 * [Akka Cluster Bootstrap](https://doc.akka.io/docs/akka-management/current/bootstrap/) helps bootstrapping an Akka cluster using Akka Discovery.
 * [Akka Management Cluster HTTP](https://doc.akka.io/docs/akka-management/current/cluster-http-management.html) provides HTTP endpoints for introspecting and managing Akka clusters.
 * [Akka Discovery for Kubernetes, Consul, Marathon, and AWS](https://doc.akka.io/docs/akka-management/current/discovery/)
+* [Kubernetes Lease](https://doc.akka.io/docs/akka-management/current/kubernetes-lease.html)
```
Comment (chbatey) [inline akka-docs/src/main/paradox/common/other-modules.md:29]: 👍 i'll move that today

**Review by chbatey — APPROVED** [review]
Looks good.

Perhaps we can have a follow up ticket to get a breakdown of which tests add the most time to the build and see if we can reduce any

**Review by johanandren — APPROVED** [review]
LGTM!

**Pull request merged**